File: src/ServerPulse.Frontend/src/app/modules/analyzer/domain/models/statistics/serverLifecycleStatistics.ts
import { BaseStatistics } from "../../..";
import { TimeSpan } from "../../../../shared";

export interface ServerLifecycleStatistics extends BaseStatistics {
    isAlive: boolean;
    dataExists: boolean;
    serverLastStartDateTimeUTC: Date | null;
    serverUptime: TimeSpan | null;
    lastServerUptime: TimeSpan | null;
    lastPulseDateTimeUTC: Date | null;
}

export function getDefaultServerLifecycleStatistics(): ServerLifecycleStatistics {
    return {
        id: "",
        collectedDateUTC: new Date(),
        isAlive: false,
        dataExists: false,
        serverLastStartDateTimeUTC: null,
        serverUptime: null,
        lastServerUptime: null,
        lastPulseDateTimeUTC: null
    }
}

File: src/ServerPulse.Backend/AnalyzerApi.Core/Models/Statistics/ServerLifecycleStatistics.cs
namespace AnalyzerApi.Core.Models.Statistics
{
    public class ServerLifecycleStatistics : BaseStatistics
    {
        public bool IsAlive { get; set; }
        public bool DataExists { get; set; }
        public DateTime? ServerLastStartDateTimeUTC { get; set; }
        public TimeSpan? ServerUptime { get; set; }
        public TimeSpan? LastServerUptime { get; set; }
        public DateTime? LastPulseDateTimeUTC { get; set; }
    }
}